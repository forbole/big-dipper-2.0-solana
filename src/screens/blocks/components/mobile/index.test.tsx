import React from 'react';
// import renderer from 'react-test-renderer';
// import { MockTheme } from '@tests/utils';
// import Mobile from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  // SingleBlockMobile: (props) => <div id="SingleBlockMobile" {...props} />,
  Loading: (props) => <div id="Loading" {...props} />,
  AvatarName: (props) => <div id="AvatarName" {...props} />,
}));

jest.mock('react-virtualized-auto-sizer', () => ({ children }: any) => children({
  height: 600, width: 600,
}));

// ==================================
// unit tests
// ==================================
describe('screen: Home/Blocks/Mobile', () => {
  // it('matches snapshot', () => {
  //   const component = renderer.create(
  //     <MockTheme>
  //       <Mobile
  //         items={[
  //           {
  //             slot: 812768640,
  //             leader: {
  //               name: 'desmosvaloper1rzhewpmmdl72lhnxj6zmxr4v94f522s4hyz467',
  //               address: 'desmosvaloper1rzhewpmmdl72lhnxj6zmxr4v94f522s4hyz467',
  //             },
  //             hash: '76nwV8zz8tLz97SBRXH6uwHvgHXtqJDLQfF66jZhQ857',
  //             txs: 2,
  //             timestamp: '2021-09-13T20:06:17.363145',
  //           },
  //           {
  //             slot: 812768640,
  //             leader: {
  //               name: 'desmosvaloper1rzhewpmmdl72lhnxj6zmxr4v94f522s4hyz467',
  //               address: 'desmosvaloper1rzhewpmmdl72lhnxj6zmxr4v94f522s4hyz467',
  //             },
  //             hash: '76nwV8zz8tLz97SBRXH6uwHvgHXtqJDLQfF66jZhQ857',
  //             txs: 2,
  //             timestamp: '2021-09-13T20:06:17.363145',
  //           },
  //         ]}
  //         itemCount={2}
  //         loadMoreItems={() => jest.fn()}
  //         isItemLoaded={() => true}
  //       />
  //     </MockTheme>,
  //   );
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  it('test', () => {
    const a = 2;
    const b = 3;
    expect(a + b).toBe(5);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
