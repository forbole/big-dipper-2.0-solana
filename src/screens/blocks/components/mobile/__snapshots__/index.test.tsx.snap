// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home/Blocks/Mobile matches snapshot 1`] = `
<div
  className="makeStyles-root"
>
  <div
    className="List"
    onScroll={[Function]}
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "direction": "ltr",
        "height": 600,
        "overflow": "auto",
        "position": "relative",
        "width": 600,
        "willChange": "transform",
      }
    }
  >
    <div
      style={
        Object {
          "height": 200,
          "pointerEvents": undefined,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "height": 100,
            "left": 0,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": "100%",
          }
        }
      >
        <div>
          <div
            hash="76nwV8zz8tLz9...F66jZhQ857"
            id="SingleBlockMobile"
            leader={
              <AvatarName
                address="desmosvaloper1rzhewpmmdl72lhnxj6zmxr4v94f522s4hyz467"
                name="desmosvaloper1rzhewpmmdl72lhnxj6zmxr4v94f522s4hyz467"
              />
            }
            slot={
              <Link
                href="/blocks/812768640"
                passHref={true}
              >
                <WithStyles(ForwardRef(Typography))
                  className="value"
                  component="a"
                  variant="body1"
                >
                  812,768,640
                </WithStyles(ForwardRef(Typography))>
              </Link>
            }
            time="1 day ago"
            txs="2"
          />
          <hr
            className="MuiDivider-root"
          />
        </div>
      </div>
      <div
        style={
          Object {
            "height": 100,
            "left": 0,
            "position": "absolute",
            "right": undefined,
            "top": 100,
            "width": "100%",
          }
        }
      >
        <div>
          <div
            hash="76nwV8zz8tLz9...F66jZhQ857"
            id="SingleBlockMobile"
            leader={
              <AvatarName
                address="desmosvaloper1rzhewpmmdl72lhnxj6zmxr4v94f522s4hyz467"
                name="desmosvaloper1rzhewpmmdl72lhnxj6zmxr4v94f522s4hyz467"
              />
            }
            slot={
              <Link
                href="/blocks/812768640"
                passHref={true}
              >
                <WithStyles(ForwardRef(Typography))
                  className="value"
                  component="a"
                  variant="body1"
                >
                  812,768,640
                </WithStyles(ForwardRef(Typography))>
              </Link>
            }
            time="1 day ago"
            txs="2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
